app/models/invoice.rb

```diff
@@ -99,14 +99,13 @@ class Invoice < ApplicationRecord
   scope :ready_to_be_refreshed, -> { where(ready_to_be_refreshed: true) }
   scope :ready_to_be_finalized,
     lambda {
-      date = <<-SQL
-            (
-              invoices.created_at +
-              COALESCE(customers.invoice_grace_period, organizations.invoice_grace_period) * INTERVAL '1 DAY'
-            )
-      SQL
-
-      draft.joins(:customer, :organization).where("#{Arel.sql(date)} < ?", Time.current)
+      draft
+        .joins(customer: :organization)
```

Comment:
no need to join anymore.

Reply:
I think it is needed for the timezone part. We need to fetch timezone there

Reply:
ah right 👍